### PR TITLE
Dilpreet | Test | Provide Fraud headers for UK company

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/auth-hmrc/auth-hmrc.component.ts [hunk 1]
```diff
@@ -40,6 +44,11 @@ export class AuthHMRCComponent implements OnInit, OnDestroy {
     * @memberof AuthHMRCComponent
     */
     public ngOnInit(): void {
+        this.generalService.getClientIp().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.ipAddress) {
+                this.clientIp = response?.ipAddress;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/auth-hmrc/auth-hmrc.component.ts]: this.clientIp = response.ipAddress;
 
Already check in if condition

File: apps/web-giddh/src/app/services/general.service.ts [hunk 2]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
     }
 
     /**
-     * This will return the system current user time zone
+     * Retrieves the operating system configuration based on the user agent string.
+     *
+     * @returns {string} The name of the operating system.
+     * @memberof GeneralService
+     */
+    public getOsConfiguration() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getOsConfiguration(): string

File: apps/web-giddh/src/app/services/general.service.ts [hunk 3]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        const userAgent = window.navigator.userAgent;
+        let osName;
+
+        if (userAgent.indexOf("Win") != -1) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: if (userAgent.indexOf("Win") !=- -1) {

use !==

File: apps/web-giddh/src/app/services/general.service.ts [hunk 4]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        return osName;
+    }
+
+    /**
+     * Retrieves the device manufacturer based on the user agent string.
+     *
+     * @returns {string} The device manufacturer.
+     * @memberof GeneralService
+     */
+    public getDeviceManufacture() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getDeviceManufacture(): string {

File: apps/web-giddh/src/app/services/general.service.ts [hunk 5]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        return deviceManufacture;
+    }
+
+    /**
+     * Retrieves the current timestamp in ISO format.
+     *
+     * @returns {string} The current timestamp.
+     * @memberof GeneralService
+     */
+    public getTimesStamp() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getTimesStamp(): string {

add return type string or any which is better

File: apps/web-giddh/src/app/services/general.service.ts [hunk 6]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        return timestamp;
+    }
+
+    /**
+     * Retrieves the device model based on the user agent string.
+     *
+     * @returns {string} The device model.
+     * @memberof GeneralService
+     */
+    public getDeviceModel() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getDeviceModel(): string {

File: apps/web-giddh/src/app/services/general.service.ts [hunk 7]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        return deviceModel;
+    }
+
+    /**
+     * Retrieves the operating system family based on the user agent string.
+     *
+     * @returns {string} The operating system family.
+     * @memberof GeneralService
+     */
+    public getOSFamily() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getOSFamily(): string {

File: apps/web-giddh/src/app/services/general.service.ts [hunk 8]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+        return osFamily;
+    }
+
+    /**
+     * Retrieves the operating system version based on the user agent string.
+     *
+     * @returns {string} The operating system version.
+     * @memberof GeneralService
+     */
+    public getOSVersion() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getOSVersion(): string {

File: apps/web-giddh/src/app/services/general.service.ts [hunk 9]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+     * This will be use for get user agent
+     *
+     * @param {*} clientIp
      * @return {*}
      * @memberof GeneralService
      */
-    public getUserTimeZone(): any {
-        let offset = new Date().getTimezoneOffset(), o = Math.abs(offset);
-        return (offset < 0 ? "+" : "-") + ("00" + Math.floor(o / 60)).slice(-2) + ":" + ("00" + (o % 60)).slice(-2);
+    public getUserAgentData(clientIp: any) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getUserAgentData(clientIp: any): any {

File: apps/web-giddh/src/app/services/general.service.ts [hunk 10]
```diff
@@ -1693,25 +1694,252 @@ export class GeneralService {
...
+     *This will return the client IP address
      *
-     * @param {*} timezone
-     * @return {*}  {*}
      * @memberof GeneralService
      */
-    public getGovClientTimezoneHeader(timezone: any): any {
-        const headers = new HttpHeaders().set('Gov-Client-Timezone', 'UTC' + timezone);
-        return { headers };
+    public getClientIp() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/general.service.ts]: public getClientIp() any {

File: apps/web-giddh/src/app/vat-report/file-return/file-return.component.ts [hunk 11]
```diff
@@ -58,6 +59,13 @@ export class FileReturnComponent implements OnInit, OnDestroy {
     * @memberof FileReturnComponent
     */
     public ngOnInit(): void {
+
+        this.generalService.getClientIp().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.ipAddress) {
+                this.clientIp = response?.ipAddress;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/file-return/file-return.component.ts]: this.clientIp = response.ipAddress;

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts [hunk 12]
```diff
@@ -105,6 +106,12 @@ export class ObligationsComponent implements OnInit, OnDestroy {
         this.isCompanyMode = this.generalService.currentOrganizationType === OrganizationType.Company;
         this.loadTaxDetails();
 
+        this.generalService.getClientIp().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.ipAddress) {
+                this.clientIp = response?.ipAddress;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts]: this.clientIp = response.ipAddress;

File: apps/web-giddh/src/app/vat-report/view-return/view-return.component.ts [hunk 13]
```diff
@@ -50,6 +54,11 @@ export class ViewReturnComponent implements OnInit {
     * @memberof ViewReturnComponent
     */
     public ngOnInit(): void {
+        this.generalService.getClientIp().pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.ipAddress) {
+                this.clientIp = response?.ipAddress;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/view-return/view-return.component.ts]: this.clientIp = response.ipAddress;

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**